dtsync/sync.go

```diff
@@ -176,6 +176,12 @@ func (s *Sync) onEvent(event dt.Event, channelState dt.ChannelState) {
 		if strings.HasSuffix(msg, "content not found") {
 			err = errors.New(err.Error() + ": content not found")
 		}
+	case dt.Ongoing:
+		if channelState.Message() != "stream reset" {
```

Comment:
will having this be an error state mean the next layer up re-starts / re-attempts sync?

Reply:
I don't know and I need @hannahhoward  to look at it.  As far as I can tell, once that is seen there is no more activity for that graphsync/datatransfer session.

Reply:
I believe this will cause the sync call to fail, so it's up to the caller to do a resync. For the indexer, we would do this next time we do a sync.

Reply:
it means the connection between the remote peer and the local peer got disconnected. It most cases the data transfer should attempt to restart the connection to get things going again, though you have to configure the channel monitor that does the restarts. When no restarts are configured one one expect it to become a failure but I didn't right the channel monitor and it sounds like it doesn't put the channel in a failure state. The SPX data transfer instance is configured for no restarts, so I would configure it on Store The Index --- you can see an example of how to do so here: https://github.com/filecoin-project/lotus/blob/master/node/modules/client.go#L120 or here -- https://github.com/application-research/filclient/blob/master/filclient.go#L186